packages/js/src/initializers/post-scraper.js

```diff
@@ -494,6 +497,26 @@ export default function initPostScraper( $, store, editorData ) {
 		// Analysis plugins
 		window.YoastSEO.wp = {};
 		window.YoastSEO.wp.replaceVarsPlugin = new YoastReplaceVarPlugin( app, store );
+		let shortcodesToBeParsed = [];
+
+		shortcodesToBeParsed = applyFilters( "yoast.analysis.shortcodes", shortcodesToBeParsed );
+
+		// Make sure the added shortcodes are valid. They are valid if they are included in `wpseo_shortcode_tags`.
+		shortcodesToBeParsed = shortcodesToBeParsed.filter( shortcode => {
+			return wpseoScriptData.analysis.plugins.shortcodes.wpseo_shortcode_tags.includes( shortcode );
+		} );
```

Comment:
Here I added a step to check is the added shortcodes are valid ones. This is done by checking if each shortcode is present in `wpseoScriptData.analysis.plugins.shortcodes.wpseo_shortcode_tags` array. In that array only shortcodes retrieved from `$GLOBALS['shortcode_tags']` are included.

Reply:
You don't need the `return` though if you're filter is one line 😸